### PR TITLE
Changes to upload consistent file UUIDs.

### PR DESCRIPTION
**Pull request by DailyDreaming**
Relies on https://github.com/HumanCellAtlas/dcp-cli/pull/502 .

Both have been tested together and appear to work.

### CODE REVIEW

File: create_project.py [hunk 1]
```diff
@@ -757,50 +757,97 @@ def parse_donor_organism_data_from_xlsx(wb):
...
+    return json_content
+
+
+def create_json_if_not_present(file_name):
+    if not os.path.exists(file_name):
+        with open(file_name, 'w') as f:
+            f.write('{}\n')
+
+
+def update_file_uuid_config_file(file_name, file_uuid, output_dir):
```
Comment (hannes-ucsc) [inline create_project.py:778]: Wouldn't this be simpler if `.upload()` accepted a `dict` with the mapping instead of a file containing the mapping?

**Review by hannes-ucsc — COMMENTED** [review]
If possible, I would like to sever the metadata generation from the uploading part. I want us to be able to say `make metadata` and `make upload` separately.

Ingest currently satisfies the invariant that `provenance.document_id` in every metadata entity is the same as the UUID of the DSS file containing that entity. Does this also satisfy that?

File: create_project.py (hunk 1 above)
Comment (DailyDreaming) [reply create_project.py:778]: I'll change it to allow both.

**Comment by DailyDreaming** [comment]
> If possible, I would like to sever the metadata generation from the uploading part. I want us to be able to say make metadata and make upload separately.

Currently upload is in two places so we'd just need to delete it here.  I'll make a PR to do so and use the Makefile from here on.

> Ingest currently satisfies the invariant that provenance.document_id in every metadata entity is the same as the UUID of the DSS file containing that entity. Does this also satisfy that?

Yes.  That's a large part of what I was trying to solve.  Our current behavior breaks that.

**Comment by hannes-ucsc** [comment]
I tried to pick this up from where @DailyDreaming left it off but found what I think to be a simpler solution, using a callback:

https://github.com/HumanCellAtlas/dcp-cli/commit/36270911697eb464262ce27b39ee4ba1668eff61

https://github.com/DailyDreaming/load-project/commit/924b576cc30304820d883875177cb76789837df7

**Pull request closed without merging**